lib/spack/env/cc

```diff
@@ -484,6 +485,25 @@ while [ $# -ne 0 ]; do
             if [ -z "$arg" ]; then shift; arg="$1"; fi
             append other_args_list "-l$arg"
             ;;
+        -Werror=*)
```

Comment:
it'd be nice to generalize these with the DTAG stripping we already do to ensure `RPATH` -- can the arguments to the wrapper not be a general set of args to strip?

Reply:
I have to think though how best to make that happen, maybe we can pass a list of patterns, and if any of them match the argument is removed? That should work as long as we don't need a conditional or other handling inside the wrapper.

Reply:
Having thought this through more, we *can* do this, in general, but `-Werror` has a bit of a trick attached to it that makes it more complicated.  If we want to be able to differentiate between `-Werror=<something>` and `-Werror`, the *longer match* has to be explicitly kept before matching the shorter match.  I'm considering whether to build out a generic enough mechanism to allow us to pass this in, or handle `-Werror=` specially and merge out everything else.